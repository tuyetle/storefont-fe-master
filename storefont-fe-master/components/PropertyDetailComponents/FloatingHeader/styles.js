import { css } from 'glamor';
import variables from '~/styles/variables';
import themes from '~/styles/themes';

const iconArrow = {
  border: `solid ${variables.$commonRed}`,
  borderWidth: '0 1px 1px 0',
  display: 'inline-block',
  padding: '0.3125em',
};

export default {
  floatingPropertyDetail: css({
    boxShadow: variables.$boxShadowBottomFloating,
    padding: '1.2em 0 0.1875em',
  }),

  wrapper: css({
    '& .priceLabel': {
      display: 'inline-block',
      textTransform: 'uppercase',
      verticalAlign: 'middle',
      '& div': {
        fontSize: '1.5625em',
      },
    },
  }),

  attributeList: css({
    display: 'inline-block',
    paddingLeft: '2.125em',
  }),

  attributeValue: css({
    fontSize: '0.9375em',
  }),

  inline: css({
    display: 'inline-block',
    paddingRight: '1.9em',

    '& label': {
      opacity: 1,

      '& span': {
        boxShadow: 'none',
      },
    },

    '& span': {
      display: 'inline-block',
      fontSize: '0.9375em',
      marginLeft: '0.125em',
    },
  }),

  widthIcon: css({
    '& span': {
      width: '1.2em',
      height: '1.2em',
      lineHeight: '1.2em',
      verticalAlign: 'middle',
      display: 'inline-block',
      marginRight: '0.8em',

      '&:nth-child(2)': {
        display: 'none',
      },
    },
  }),

  saveWrap: css({
    ...themes['font-montserrat-medium'],
    display: 'inline-block',
    verticalAlign: 'top',
    marginTop: '0.6em',

    '& .btn.btn-link': {
      textDecoration: 'none',
      color: `${variables.$listingItemTitleColor}`,
      fontWeight: 400,
      textTransform: 'uppercase',
      fontSize: '1em',
      lineHeight: '1em',
    },
    '& .btn.btn-link:hover': {
      textDecoration: 'none',
      color: `${variables.$listingItemTitleColor}`,
    },
    '& svg': {
      float: 'left',
    },
    '& span': {
      display: 'inline-block',
      paddingLeft: '0.5em',
    },
    hearIcon: {
      position: 'relative',
      top: '0.3em',
    },
  }),

  submitButtonContainer: css({
    display: 'inline-block',
    marginLeft: '2.5em',
  }),

  btnRequest: css({
    ...themes['button-red'],
    ...themes['font-montserrat-medium'],
  }),

  iconSend: css({
    marginRight: '0.1875em',
  }),

  openForm: css({
    textAlign: 'center',
    marginTop: '-2px',
  }),

  iconOpenDown: css({
    ...iconArrow,
    transform: 'rotate(45deg)',
    WebkitTransform: 'rotate(45deg)',
  }),
};

