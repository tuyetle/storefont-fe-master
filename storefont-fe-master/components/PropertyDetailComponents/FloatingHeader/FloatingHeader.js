import React, { PureComponent } from 'react';
import { translate } from 'react-i18next';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import AddressPropertyDetail from '~/components/PropertyDetailComponents/HeaderPropertyDetail/AddressPropertyDetail/AddressPropertyDetail';
import PriceLabelPropertyDetail from '~/components/PropertyDetailComponents/HeaderPropertyDetail/PriceLabelPropertyDetail/PriceLabelPropertyDetail';
import IconUtilityDisplaying from '~/components/IconUtility/IconUtilityDisplaying';
import SquareAttribute from '~/components/PropertyDetailComponents/AttributeValues/SquareAttribute';
import FloatingContactButton from '~/components/PropertyDetailComponents/FloatingContactButton/FloatingContactButton';
import HeartIcon from '~/static/assets/img/heart.svg';
import styles from './styles';

const AREA_TYPE = 'area';

export class FloatingHeader extends PureComponent {
  render() {
    const {
      t, attributes, address, transactionType, priceText,
      postContact, postContactResult, requestContact,
      onRequestContactChanged,
    } = this.props;

    const FavoritedIcon = HeartIcon;

    const listItems = attributes.map(item =>
      (
        <div className={`${styles.inline}`} key={item.name}>
          <IconUtilityDisplaying
            name={item.name}
            className={`${styles.widthIcon}`}
          />
          {item.name === AREA_TYPE ?
            <SquareAttribute item={item} />
            :
            (<span className={`${styles.attributeValue}`}>{item.value} </span>)
          }
        </div>
      ));

    return (
      <div className={styles.floatingPropertyDetail}>
        <div className="container">
          <div className={classNames('row', `${styles.wrapper}`)}>
            <div className="col-12 col-md-7">
              <AddressPropertyDetail
                address={address}
                transactionType={transactionType}
              />
              <PriceLabelPropertyDetail
                priceText={priceText}
              />
              <div className={styles.attributeList}>
                {listItems}
              </div>
            </div>
            <div className="col-12 col-md-2 text-right">
              {
                <div className={styles.saveWrap}>
                  <button className="btn btn-link">
                    <FavoritedIcon className={styles.hearIcon} />
                    <span>{t(`common:saveListing`)}</span>
                  </button>
                </div>
              }
            </div>
            <div className="col-12 col-md-3 text-right">
              {
                this.props.isShowButtonContact &&
                <FloatingContactButton
                  postContact={postContact}
                  postContactResult={postContactResult}
                  requestContact={requestContact}
                  onRequestContactChanged={onRequestContactChanged}
                />
              }
            </div>
          </div>
        </div>
      </div>
    );
  }
}

FloatingHeader.propTypes = {
  t: PropTypes.func,
  address: PropTypes.string.isRequired,
  transactionType: PropTypes.string.isRequired,
  priceText: PropTypes.number,
  attributes: PropTypes.arrayOf(PropTypes.object).isRequired,
  isShowButtonContact: PropTypes.bool,
  postContact: PropTypes.func,
  postContactResult: PropTypes.shape({
    statusCode: PropTypes.number,
    message: PropTypes.string,
  }),
  onRequestContactChanged: PropTypes.func,
  requestContact: PropTypes.object,
};

FloatingHeader.defaultProps = {
  t: null,
  priceText: 0,
  isShowButtonContact: false,
  postContact() {},
  postContactResult: null,
  onRequestContactChanged: null,
  requestContact: null,
};

export default translate(['propertyDetail', 'common'])(FloatingHeader);
