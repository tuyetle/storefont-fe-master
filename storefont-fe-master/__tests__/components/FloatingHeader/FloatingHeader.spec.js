import React from 'react';
import * as renderer from 'react-test-renderer';
import FloatingHeader from '~/components/PropertyDetailComponents/FloatingHeader/FloatingHeader';

describe('(Component) FloatingHeader', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      address: 'An Dương Vương, Phường Phú Thượng, Quận Tây Hồ',
      transactionType: 'Bán',
      priceText: 150000,
      attributes: [
        {
          name: 'area', displayName: 'Diện tích', value: 65, addon: 'm2',
        },
        {
          name: 'bedrooms', displayName: 'Phòng ngủ', value: 3,
        },
        {
          name: 'bathrooms', displayName: 'Phòng tắm', value: 2,
        },
      ],
    };
    component = mount(<FloatingHeader {...props} />);
  });

  it('should renders correctly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

