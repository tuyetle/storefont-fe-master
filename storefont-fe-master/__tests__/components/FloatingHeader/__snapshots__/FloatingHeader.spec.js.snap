// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) FloatingHeader should renders correctly 1`] = `
<div
  className={
    Object {
      "data-css-y1wi13": "",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="row css-821ecb"
    >
      <div
        className="col-12 col-md-7"
      >
        <div
          className={
            Object {
              "data-css-fisw11": "",
            }
          }
        >
          <svg
            className={
              Object {
                "data-css-yhgus": "",
              }
            }
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="#686868"
            >
              <path
                d="M255.5 111.5c-35.2 0-63.8 28.6-63.8 63.8 0 35.2 28.6 63.8 63.8 63.8 35.1 0 63.8-28.6 63.8-63.8 0-35.2-28.6-63.8-63.8-63.8zm0 109.3c-25.1 0-45.6-20.4-45.6-45.6 0-25.1 20.4-45.6 45.6-45.6 25.1 0 45.6 20.5 45.6 45.6 0 25.2-20.5 45.6-45.6 45.6zm0 0"
              />
              <path
                d="M377.5 53.3c-33-33-76.9-51.2-123.5-51.2-46.7 0-90.6 18.2-123.6 51.2-61 61.1-68.6 175.9-16.4 245.6L254 501l139.7-201.9c52.4-69.8 44.8-184.7-16.2-245.8zm1.4 235.2L253.9 469 128.7 288.3c-47.3-63.2-40.6-166.9 14.5-222 29.6-29.5 68.9-45.8 110.7-45.8S335 36.8 364.5 66.3c55.2 55.1 62 158.8 14.4 222.2zm0 0"
              />
            </g>
          </svg>
          <span
            className={
              Object {
                "data-css-1awbpa": "",
              }
            }
          >
            An Dương Vương, Phường Phú Thượng, Quận Tây Hồ
          </span>
          <span
            className={
              Object {
                "data-css-vfobqd": "",
              }
            }
          >
            Bán
          </span>
        </div>
        <div
          className="priceLabel"
        >
          <div
            className={
              Object {
                "data-css-tl6b3y": "",
              }
            }
          />
          <div
            className={
              Object {
                "data-css-1hlt6jn": "",
              }
            }
          >
            
          </div>
        </div>
        <div
          className={
            Object {
              "data-css-f0rhr9": "",
            }
          }
        >
          <div
            className="css-51zxzb"
          >
            <label
              className="css-d32kd0 css-rr0qee "
            >
              <span
                className="css-d32kd0 css-18ndbk"
              >
                <svg
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M512 129.3V-.4H382.2v56.9H129.3V-.4H-.4v129.7h56.9v252.9H-.4V512h129.7v-56.9h252.9V512H512V382.2h-56.9V129.3H512zM398.1 15.5h98v97.9h-98V15.5zM15.5 113.4V15.5h97.9v98l-97.9-.1zm98 382.7h-98v-98h98v98zm382.6-98v98h-98v-98h98zm-56.9-15.9h-56.9v56.9h-253v-56.9H72.4V129.3h56.9V72.4h252.9v56.9h56.9v252.9zm0 0"
                    fill="#686868"
                  />
                </svg>
              </span>
              <span
                className={
                  Object {
                    "data-css-nuruvz": "",
                  }
                }
              >
                
              </span>
            </label>
            <span
              className={
                Object {
                  "data-css-qww5z6": "",
                }
              }
            >
              65
              <span>
                 m
                <sup>
                  2
                </sup>
              </span>
            </span>
          </div>
          <div
            className="css-51zxzb"
          >
            <label
              className="css-d32kd0 css-rr0qee "
            >
              <span
                className="css-d32kd0 css-18ndbk"
              >
                <svg
                  id="Layer_1"
                  viewBox="0 0 512 321.2"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <style>
                    .st0
                    {
                    fill:#686868
                    }
                  </style>
                  <path
                    className="st0"
                    d="M505.3 321.2h-38.2c-3.7 0-6.7-3-6.7-6.7v-49.2H54.7v49.2c0 3.7-3 6.7-6.7 6.7H6.7c-3.7 0-6.7-3-6.7-6.7V6.7C0 3 3 0 6.7 0H48c3.7 0 6.7 3 6.7 6.7v91.5c4.9-3.4 10.9-5.4 17.3-5.4h82.7c16.9 0 30.7 13.8 30.7 30.7v8.9c0 6.4-2 12.4-5.4 17.4h280.4V88.9c0-3.7 3-6.7 6.7-6.7h38.2c3.7 0 6.7 3 6.7 6.7v225.6c0 3.7-3 6.7-6.7 6.7zm-31.5-13.3h24.9V95.6h-24.9v212.3zm-460.5 0h28V13.3h-28v294.6zM54.7 252h405.8v-23.6H54.7V252zm0-36.9h405.8v-52H54.7v52zM72 149.7h82.7c9.6 0 17.3-7.8 17.3-17.3v-8.9c0-9.6-7.8-17.3-17.3-17.3H72c-9.6 0-17.3 7.8-17.3 17.3v8.9c0 9.5 7.8 17.3 17.3 17.3z"
                  />
                </svg>
              </span>
              <span
                className={
                  Object {
                    "data-css-nuruvz": "",
                  }
                }
              >
                
              </span>
            </label>
            <span
              className="css-wk70fd"
            >
              3
               
            </span>
          </div>
          <div
            className="css-51zxzb"
          >
            <label
              className="css-d32kd0 css-rr0qee "
            >
              <span
                className="css-d32kd0 css-18ndbk"
              >
                <svg
                  viewBox="-49 141 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M437.4 386.6H-23.4c-14.5 0-25.6-11.1-25.6-25.6 0-14.5 11.1-25.6 25.6-25.6h460.8c14.5 0 25.6 11.1 25.6 25.6 0 14.5-11.1 25.6-25.6 25.6zm-460.8-34.2c-5.1 0-8.5 3.4-8.5 8.5s3.4 8.5 8.5 8.5h460.8c5.1 0 8.5-3.4 8.5-8.5s-3.4-8.5-8.5-8.5H-23.4z"
                    fill="#686868"
                  />
                  <path
                    d="M346.1 523.1H67.9c-27.3 0-51.2-18.8-58-45.2l-24.7-98.1c-.9-2.6 0-5.1 1.7-7.7 1.7-1.7 4.3-2.6 6.8-2.6h426.7c2.6 0 5.1.9 6.8 3.4 1.7 1.7 2.6 5.1 1.7 7.7l-24.7 98.1c-6.9 25.6-30.8 44.4-58.1 44.4zM4.8 386.6l22.2 87c5.1 18.8 22.2 32.4 41.8 32.4H347c19.6 0 36.7-13.7 41.8-32.4l20.5-87H4.8z"
                    fill="#686868"
                  />
                  <path
                    d="M70.5 557.2c-.9 0-2.6 0-3.4-.9-4.3-1.7-6-6.8-3.4-11.1l17.1-34.1c1.7-4.3 6.8-6 11.1-3.4 4.3 1.7 6 6.8 3.4 11.1L78.1 553c-1.7 2.5-4.2 4.2-7.6 4.2zM352.1 557.2c-3.4 0-6-1.7-7.7-5.1L327.3 518c-1.7-4.3 0-9.4 3.4-11.1 4.3-1.7 9.4 0 11.1 3.4l17.1 34.1c1.7 4.3 0 9.4-3.4 11.1-.9 1.7-1.7 1.7-3.4 1.7zM428.9 336.5h-17.1V211.1c0-30.4-26.5-54.6-59.7-54.6-35 0-51.2 27.3-51.2 54.6h-17.1c0-41.3 28.2-70.1 68.3-70.1 42.7 0 76.8 31.2 76.8 70.1v125.4zM258.2 283.2c-5.1 0-8.5-3.1-8.5-7.8v-3.9c0-4.7 3.4-7.8 8.5-7.8s8.5 3.1 8.5 7.8v3.9c0 4.7-3.4 7.8-8.5 7.8zM258.2 319c-5.1 0-8.5-3.1-8.5-7.8v-8.6c0-4.7 3.4-7.8 8.5-7.8s8.5 3.1 8.5 7.8v8.6c0 3.9-3.4 7.8-8.5 7.8zM292.3 283.2c-5.1 0-8.5-3.1-8.5-7.8v-3.9c0-4.7 3.4-7.8 8.5-7.8s8.5 3.1 8.5 7.8v3.9c.1 4.7-3.3 7.8-8.5 7.8zM292.3 319c-5.1 0-8.5-3.1-8.5-7.8v-8.6c0-4.7 3.4-7.8 8.5-7.8s8.5 3.1 8.5 7.8v8.6c.1 3.9-3.3 7.8-8.5 7.8zM326.5 283.2c-5.1 0-8.5-3.1-8.5-7.8v-3.9c0-4.7 3.4-7.8 8.5-7.8s8.5 3.1 8.5 7.8v3.9c0 4.7-3.4 7.8-8.5 7.8zM326.5 319c-5.1 0-8.5-3.1-8.5-7.8v-8.6c0-4.7 3.4-7.8 8.5-7.8s8.5 3.1 8.5 7.8v8.6c0 3.9-3.4 7.8-8.5 7.8z"
                    fill="#686868"
                  />
                  <path
                    d="M326.5 248.1h-68.3c-5.1 0-8.5-3.1-8.5-7.8 0-21.8 18.8-39 42.7-39 23.9 0 42.7 17.1 42.7 39-.1 4.7-3.5 7.8-8.6 7.8zm-58.1-15.6H317c-3.4-9.4-12.8-15.6-23.9-15.6s-21.2 6.3-24.7 15.6z"
                    fill="#686868"
                  />
                </svg>
              </span>
              <span
                className={
                  Object {
                    "data-css-nuruvz": "",
                  }
                }
              >
                
              </span>
            </label>
            <span
              className="css-wk70fd"
            >
              2
               
            </span>
          </div>
        </div>
      </div>
      <div
        className="col-12 col-md-2 text-right"
      >
        <div
          className={
            Object {
              "data-css-1gqq8rq": "",
            }
          }
        >
          <button
            className="btn btn-link"
          >
            <svg
              className={undefined}
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.745.39A4.204 4.204 0 0 0 7.5 1.927 4.201 4.201 0 0 0 4.254.39 4.208 4.208 0 0 0 .051 4.593c0 1.814 1.082 3.914 3.216 6.238 1.642 1.789 3.428 3.175 3.937 3.557l.296.222.295-.223c.508-.382 2.294-1.768 3.936-3.557 2.136-2.323 3.217-4.424 3.217-6.238A4.207 4.207 0 0 0 10.745.39zm.264 9.777c-1.351 1.473-2.806 2.661-3.509 3.21-.705-.549-2.158-1.737-3.508-3.21C2.057 8.06 1.034 6.131 1.034 4.593a3.224 3.224 0 0 1 3.221-3.22A3.22 3.22 0 0 1 7.07 3.034l.43.776.43-.775a3.222 3.222 0 0 1 2.815-1.661 3.224 3.224 0 0 1 3.222 3.22c0 1.537-1.024 3.466-2.958 5.573z"
                fill="#404141"
              />
            </svg>
            <span>
              
            </span>
          </button>
        </div>
      </div>
      <div
        className="col-12 col-md-3 text-right"
      />
    </div>
  </div>
</div>
`;
